progress/__init__.py

```diff
@@ -95,8 +96,8 @@ def __init__(self, *args, **kwargs):
 
     @property
     def eta(self):
-        avg = self.avg
-        return int(ceil(self.remaining / avg)) if avg else 0
+        if not self.index: return 0
+        return int(ceil((self.max - self.index) * self.elapsed/self.index))
```

Comment:
Why you are not using `self.remaining` here? It has an extra bounds check. Also, with this change you are basically disabling the moving average algorithm and use plain average.